src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,216 @@
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_CERTIFICATE;
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_PRIVATE_KEY;
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_PUBLIC_KEY;
+import static com.aws.greengrass.ipc.common.ExceptionUtil.translateExceptions;
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.SUBSCRIBE_TO_CERTIFICATE_UPDATES;
+
+public class SubscribeToCertificateUpdatesOperationHandler
+        extends GeneratedAbstractSubscribeToCertificateUpdatesOperationHandler {
+    private static final Logger logger = LogManager.getLogger(SubscribeToCertificateUpdatesOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String NO_CERT_OPTIONS_ERROR = "Certificate options are required.";
+    private static final String INVALID_CERT_TYPE_ERROR = "Invalid certificate type.";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final CertificateManager certificateManager;
+    private final AtomicBoolean subscriptionResponseSent = new AtomicBoolean(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingEvent = new AtomicReference<>(null);
+    private final AuthorizationHandler authorizationHandler;
+    private KeyPair keyPair;
+    Consumer<X509Certificate> serverCertificateCallback = this::forwardServerCertificates;
+
+    /**
+     * Constructor.
+     *
+     * @param context               operation continuation handler
+     * @param certificateManager certificate manager
+     * @param authorizationHandler authorization handler
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         CertificateManager certificateManager,
+                                                         AuthorizationHandler authorizationHandler) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.certificateManager = certificateManager;
+        this.authorizationHandler = authorizationHandler;
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        certificateManager.unsubscribeFromServerCertificateUpdates(serverCertificateCallback);
+    }
+
+    /**
+     * Override to appropriately enforce stream responses are sent after the initial response.
+     * This only gets called if handleRequest returns normally and starts to send a response.
```

Comment:
Don't copy docs from interface (assuming this is from the interface)

Reply:
Right. Those lines are taken from interface. Will update.